From: Hans-Joachim Picht <hpicht@redhat.com>
Date: Mon, 10 Dec 2007 14:06:01 +0100
Subject: [s390] cio: add missing reprobe loop end statement
Message-id: 20071210130601.GG29540@redhat.com
O-Subject: [RHEL5 U2 PATCH 7/8] s390 - cio: add missing reprobe loop end statement
Bugzilla: 412891
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
============

Add loop end statement to prevent looping over empty subchannel sets.
As a result, CPU consumption of device probing is significantly
reduced.

Bugzilla
=========

BZ 412891
https://bugzilla.redhat.com/show_bug.cgi?id=412891

Upstream status of the patch:
=============================
Patch included in git as commit
671756162cfb0b3ccbb6a0047baa3010885561a2

Test status:
============
Kernel with patch was built and successfully tested

Please ACK.

With best regards,

Hans

 drivers/s390/cio/css.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/s390/cio/css.c b/drivers/s390/cio/css.c
index 8cc2653..0e04fd3 100644
--- a/drivers/s390/cio/css.c
+++ b/drivers/s390/cio/css.c
@@ -388,6 +388,7 @@ static int reprobe_subchannel(struct subchannel_id schid, void *data)
 		break;
 	case -ENXIO:
 	case -ENOMEM:
+	case -EIO:
 		/* These should abort looping */
 		break;
 	default:
